### ISSUE TITLE
[consistent-type-import] when the jsx element is used as a type too, it gets detected as type-only use (preact)

### ISSUE THREAD

**Issue by Haroenv** [typescript-eslint/typescript-eslint#3749]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-import": ["error"]
  }
}
```

```TS
/** @jsx h */

import { h } from 'preact';

type Props = {
  children?: h.JSX.Element;
};

export default (props: Props) => <pre>{JSON.stringify(props)}</pre>;
```

**Expected Result**

Nothing changed

**Actual Result**

```
Import "h" is only used as types
@typescript-eslint/consistent-type-imports
```

autofix moves h to a type

**Additional Info**

- No exception thrown
- does not happen for function components, or simple classes
- only happens when h is used as a type as well

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.1` and `4.29.2` |
| `@typescript-eslint/parser`        | `4.15.1` and `4.29.2` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.18.3` |

**Comment by Haroenv** [comment]
similar issues that seem to be different: https://github.com/typescript-eslint/typescript-eslint/issues/2455, https://github.com/typescript-eslint/typescript-eslint/issues/2545

big PR in our project that enables this rule (and causes errors) https://github.com/algolia/instantsearch.js/pull/4837

**Comment by Haroenv** [comment]
The solution in my case was changing from `h.JSX.something` to `JSX.something` as I realised that's also exported

**Comment by bradzacher** [comment]
You didn't include an eslint config, so IDK what you've set.
But you need to tell us that you're using preact, or else we'll just assume react.

https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#parseroptionsjsxpragma

**Comment by Haroenv** [comment]
That makes sense, from the docs I thought it would take from the tsconfig, which is set up with the jsxPragma. 

Thanks for your help!

**Comment by bradzacher** [comment]
If you're using type-aware linting, we will use whatever you've set in the tsconfig.

But without type-aware linting we don't know know how to read your tsconfig.
